src/python/pants/option/global_options.py

```diff
@@ -198,16 +198,30 @@ def from_options(cls, options: Options) -> ExecutionOptions:
                     )
                     remote_instance_name = auth_plugin_result.instance_name
 
-        # Prefix the remote servers with a scheme.
+        # Determine the remote servers.
+        # NB: Tonic expects the schemes `http` and `https`, even though they are gRPC requests.
+        # We have users set `grpc` and `grpcs` in the options system for clarity, but then
+        # normalize to `http`/`https`.
         remote_address_scheme = "https://" if bootstrap_options.remote_ca_certs_path else "http://"
-        remote_execution_address = (
-            f"{remote_address_scheme}{bootstrap_options.remote_execution_server}"
-            if bootstrap_options.remote_execution_server
-            else None
-        )
-        remote_store_addresses = [
-            f"{remote_address_scheme}{addr}" for addr in bootstrap_options.remote_store_server
-        ]
+        if bootstrap_options.remote_execution_address:
+            remote_execution_address = bootstrap_options.remote_execution_address.replace(
+                "grpc", "http"
```

Comment:
This will also replace `"grpc"` in the middle of a DNS name as well:

```
>>> "grpcs://grpc-endpoint.example.com".replace("grpc", "http")
'https://http-endpoint.example.com'
```

This should only replace the prefix at the start of the string with `http://` / `https://`, not every occurrence.

Reply:
Fixed and tweaked the test to check this.